clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -432,7 +436,7 @@ private void commitSyncWithRetries(OffsetCommitRequestState requestAttempt,
                 result.complete(null);
             } else {
                 if (error instanceof RetriableException) {
-                    if (error instanceof TimeoutException && requestAttempt.isExpired) {
+                    if (error instanceof TimeoutException && requestAttempt.isExpired()) {
```

Comment:
Why do we only check expiration in case of a `TimeoutException` but not for other retriable exceptions?
The `TopicMetadataRequestManager` handles this differently.

Reply:
I've updated the logic that checks `isExpired()`. PTAL at the latest version. Thanks!